app/controllers/api/automate_domains_controller.rb

```diff
@@ -1,5 +1,24 @@
 module Api
   class AutomateDomainsController < BaseController
+    def delete_resource(type, id = nil, _data = {})
+      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        domain = resource_search(id, type, klass)
+        api_log_info("Deleting #{automate_domain_ident(domain)}")
+
+        begin
+          # Do not delete the domain if any are locked
+          MiqAeDomain.where(:name => domain.name).each { |d| raise "Not deleting. Domain is locked." if d.contents_locked? }
```

Comment:
@jvlcek Looks good, you might want to mention somewhere that we won't be deleting
* System Domains
* GIT Based domains

The only ones you can delete are user domains which are unlocked.

Reply:
Will do. Thank you.